internal/ignition/ignition.go

```diff
@@ -744,6 +746,17 @@ func (g *installerGenerator) addStaticIPsConfigToIgnition(ignition string) error
 	return nil
 }
 
+func (g *installerGenerator) addCarrierWaitTimeoutInIgnition(ignition string) error {
+	path := filepath.Join(g.workDir, ignition)
+	config, err := parseIgnitionFile(path)
+	if err != nil {
+		return err
+	}
+
+	setFileInIgnition(config, "/etc/NetworkManager/conf.d/carrier-timeout.conf", fmt.Sprintf("data:,%s", url.PathEscape(carrierWaitTimeout)), false, 0o644)
```

Comment:
the files under /etc/NetworkManager/conf.d are executed in aphabetical order. Currently it won't matter right now, but it think it is better to give the file a number( lfor example /etc/NetworkManager/conf.d/02-carrier-timeout.conf), in case it will be needed later

Reply:
Thanks, updated